spark/v3.3/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -73,6 +73,7 @@ statement
     | ALTER TABLE multipartIdentifier WRITE writeSpec                                       #setWriteDistributionAndOrdering
     | ALTER TABLE multipartIdentifier SET IDENTIFIER_KW FIELDS fieldList                    #setIdentifierFields
     | ALTER TABLE multipartIdentifier DROP IDENTIFIER_KW FIELDS fieldList                   #dropIdentifierFields
+    | ALTER TABLE multipartIdentifier CREATE TAG identifier (AS OF VERSION snapshotId)?  (RETAIN snapshotRefRetain snapshotRefRetainTimeUnit)? #createTag
```

Comment:
I am debating with myself, if we should merge this case with the CREATE BRANCH case or not. 

I guess this question will also come for REPLACE BRANCH.

Currently leaning towards keep it as is so it is more clear, each extension just does 1 exact thing, similar to we have 2 different actions for SET/DROP identifier field, 3 for ADD/DROP/REPLACE partition field. 

Any thoughts? @amogh-jahagirdar @hililiwei

Reply:
I think it make sense just to keep it separate as well. It's nice just to read the sql extension and immediately see that CREATE TAG only supports RETAIN (there's no notion of min snapshots to keep). Same principle applies with CREATE BRANCH (can see what are the clauses that can be applied). Otherwise one has to read the actual action logic to verify that.

Reply:
Agree with @amogh-jahagirdar.